flexmeasures/api/v1_3/implementations.py

```diff
@@ -143,12 +144,31 @@ def get_device_message_response(generic_asset_name_groups, duration):
                 return unknown_schedule(
                     message + f'no data is known from "{schedule_data_source_name}".'
                 )
+            # Subquery to get the most recent schedule only
+            subq = (
+                db.session.query(
+                    Power.datetime,
+                    Power.data_source_id,
+                    func.min(Power.horizon).label("most_recent_belief_horizon"),
```

Comment:
I can't figure out why we need this in a subquery instead of adding it to the main query.

Reply:
I have no idea how this would work without a subquery. If you use `func.min` in the main query, we get SQLAlchemy complaining about two columns we want to have included in the results (`cls.cumulative_probability` and `cls.event_value`), specifically, that they `must appear in the GROUP BY clause or be used in an aggregate function`, neither of which is what we want. I don't see another way besides joining two queries.

For reference, https://github.com/SeitaBV/timely-beliefs/pull/76 is where we first developed this.